src/Components/Users/UserAdd.tsx

```diff
@@ -134,6 +135,24 @@ export const UserAdd = (props: UserProps) => {
   >([]);
   const [phoneIsWhatsApp, setPhoneIsWhatsApp] = useState(true);
   const [usernameInputInFocus, setUsernameInputInFocus] = useState(false);
+  const [usernameInput, setUsernameInput] = useState("");
+  const [usernameExists, setUsernameExists] = useState<number>(0);
+
+  const checkUsername = async (username : string) => {
+    setUsernameExists(1);
+    const userDetails = await dispatchAction(getUserDetails(username), true);
+    setUsernameExists(userDetails.status === 404 ? 3 : 2);
+  }
+
+  useEffect(()=>{
+    setUsernameExists(0);
+    if(usernameInput.length > 1 && !(state.form.username?.length < 2) && /[^.@+_-]/.test(state.form.username[state.form.username?.length - 1])){
+      let timeout = setTimeout(() => {
+        checkUsername(usernameInput);
+      }, 500);
+      return ()=>clearTimeout(timeout);
```

Comment:
instead of directly using setTimeout can you use debounce

refer: https://github.com/coronasafe/care_fe/blob/develop/src/Components/Patient/PatientRegister.tsx#L858